### DISCUSSION TITLE
How exactly Ventoy call X:/VentoyAutoRun.bat with ISOPATH and MOUNTDRIVE parameters ?

### DISCUSSION THREAD

**Discussion by devdevadev** [ventoy/Ventoy#1153] [Q&A]
Hi LomgPanda...

Which .cmd / .bat  of Ventoy actually call X:/VentoyAutoRun.bat with ISOPATH and MOUNTDRIVE parameters ?

Can you please explain the process after replacing original winpeshl.exe with your modified winpeshl.exe ?

Regards

**Comment by ventoy** [comment]
All the details here:
https://github.com/ventoy/Ventoy/blob/master/vtoyjump/vtoyjump/vtoyjump.c

**Reply by devdevadev** [reply]
Those codes are beyond my IQ...

I want to know what exact .exe , .cmd , .ini , .bat files are executing within WiinPE environment when we boot Windows ISO using Ventoy ? And in the process which file is calling X:/VentoyAutoRun.bat ?

**Reply by crasadure** [reply]
you can watch my video about VentoyAutoRun.bat 
ı captured it for turkish users...may be you can undarstand usage it :)
Video Link: https://youtu.be/bwYkM4iaC6o

**Comment by steve6375** [comment]
You don't need to worry about it! As I understand it...
When Windows/WinPE boots it loads a shell - usually WinPEShl.exe or PECMD.exe.
Ventoy replaces this .exe file with its own version which  loads the ISO file as a virtual DVD, calls the VentoyAutoRun.bat file (if it exists) and then runs the original WinPEShl.exe or PECMD.exe.
In this way the mounting of the ISO and the calling of the batch file is transparent and does not interfere with the running of whateved shell has been used and whatever script is then run by that shell.
